notebooks/utils/estimation.py

```diff
@@ -0,0 +1,229 @@
+        n_iter_max: max iteration allowed for potato convergence
+        pos_label: label for accepted trials in predict
+        neg_label: label for rejected trials in predict
+    """
+
+    def __init__(self, classes: tuple = None, potato_threshold: float = .1, field_threshold: float = .1,
+                 n_iter_max: int = 100, pos_label=1, neg_label=0):
+
+        self.classes = classes
+        self.potato_threshold = potato_threshold
+        self.field_threshold = field_threshold
+        self.n_iter_max = n_iter_max
+        self.pos_label = pos_label
+        self.neg_label = neg_label
+
+    def fit(self, X: np.array, y: np.array) -> object:
+        """Fit.
+
+        Do nothing. For compatibility purpose.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_samples)
+            ndarray of trials.
+        y : ndarray shape (n_trials,)
+            labels corresponding to each trial, not used.
+
+        Returns
+        -------
+        self : SSVEPPotato instance
+            The SSVEPPotato instance.
+        """
+        self.classes = self.classes or set(y)
+        self._n_potatoes = len(self.classes)
+        self._potatoes = []
+        for _class in self.classes:
+            X_class = X[y == _class, :, :]
+            potato = Potato(n_iter_max=self.n_iter_max, threshold=self.potato_threshold).fit(X_class)
+            self._potatoes.append(potato)
+
+        return self
+
+    def transform(self, X: np.array) -> np.array:
+        """return the SQI, combining the output z-scores of the potatoes
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_channels)
+            ndarray of SPD matrices.
+
+        Returns
+        -------
+        p : ndarray, shape (n_epochs, 1)
+         p ∈ [0, 1] is the SQI, combining the output z-scores of the potatoes.
+        """
+
+        # See equations 15, 16, 17 of paper:<https://hal.archives-ouvertes.fr/hal-02015909/document>
+        z_to_p = lambda z_score: 1 - norm.cdf(z_score)
+        q_to_p = lambda p_value: 1 - chi2.cdf(p_value, df=self._n_potatoes)
+        n_trials, n_channels, n_samples = X.shape
+
+        q = np.zeros(n_trials)
+        for potato in self._potatoes:
+            z_score = potato.transform(X)
+            q += np.log(np.apply_along_axis(z_to_p, axis=0, arr=z_score))
+        p = np.apply_along_axis(q_to_p, axis=0, arr=2 * q)
+        return p
+
+    def predict(self, X):
+        """predict artifacts from data.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_channels)
+            ndarray of SPD matrices.
+
+        Returns
+        -------
+        pred : ndarray of bool, shape (n_epochs, 1)
+            the artifact detection. True if the trial is clean, and False if
+            the trial contain an artifact.
+        """
+        p = self.transform(X)
+        pred = p < self.field_threshold
```

Comment:
This must be : `pred = p > self.field_threshold`, 
because an epoch is considered as normal/clean for a high p-value `p`, and abnormal/artifacted for a low value of `p`.